File: backend/app/models/focus_group.py
from pydantic import BaseModel
from typing import Optional


class FocusGroupRequest(BaseModel):
    product_name: str
    price: str
    currency: str = "USD"
    target_audience: str
    description: str


class AgentTranscript(BaseModel):
    skeptic: str
    bargain_hunter: str
    target_fan: str


class JudgeVerdict(BaseModel):
    summary: str
    biggest_red_flag: str
    biggest_selling_point: str
    launch_score: int


class FocusGroupResponse(BaseModel):
    transcript: AgentTranscript
    verdict: JudgeVerdict
    raw_verdict: Optional[str] = None

File: frontend/src/types/focusGroup.ts
export interface FocusGroupInput {
  product_name: string;
  price: string;
  currency: string;
  target_audience: string;
  description: string;
}

export interface AgentTranscript {
  skeptic: string;
  bargain_hunter: string;
  target_fan: string;
}

export interface JudgeVerdict {
  summary: string;
  biggest_red_flag: string;
  biggest_selling_point: string;
  launch_score: number;
}

export interface FocusGroupResult {
  transcript: AgentTranscript;
  verdict: JudgeVerdict;
  raw_verdict?: string;
}
